code/modules/watchtower/watchtower.dm

```diff
@@ -0,0 +1,381 @@
+#define WATCHTOWER_STAGE_HEIGHTNED_WELDER 2.5
+#define WATCHTOWER_STAGE_HEIGHTNED_WRENCH 3
+#define WATCHTOWER_STAGE_FLOOR 4
+#define WATCHTOWER_STAGE_BARRICADED 5
+#define WATCHTOWER_STAGE_ROOF_SUPPORT 6
+#define WATCHTOWER_STAGE_COMPLETE 7
+
+/obj/structure/watchtower
+	name = "watchtower"
+	icon = 'icons/obj/structures/watchtower.dmi'
+	icon_state = "stage1"
+
+	density = FALSE
+	bound_width = 64
+	bound_height = 96
+	health = 1000
+	layer = ABOVE_TURF_LAYER
+	var/max_health = 1000
+
+	var/stage = 1
+	var/image/roof_image
+
+/obj/structure/watchtower/Initialize()
+	var/list/turf/blocked_turfs = CORNER_BLOCK(get_turf(src), 2, 1) + CORNER_BLOCK_OFFSET(get_turf(src), 2, 1, 0, 2)
+
+	var/atom/west_blocker = new /obj/structure/blocker/watchtower(locate(x, y+1, z))
+	var/atom/east_blocker = new /obj/structure/blocker/watchtower(locate(x+1, y+1, z))
+	west_blocker.dir = WEST
+	east_blocker.dir = EAST
+
+	for(var/turf/current_turf in blocked_turfs)
+		new /obj/structure/blocker/watchtower/full_tile(current_turf)
+
+	update_icon()
+
+	return ..()
+
+/obj/structure/watchtower/Destroy()
+	playsound(src, 'sound/effects/metal_crash.ogg', 50, 1)
+	var/list/turf/top_turfs = CORNER_BLOCK_OFFSET(get_turf(src), 2, 1, 0, 1)
+
+	for(var/turf/current_turf in top_turfs)
+		for(var/mob/falling_mob in current_turf.contents)
+			falling_mob.ex_act(100, 0)
+			on_leave(falling_mob)
+
+	new /obj/structure/girder(get_turf(src))
+	new /obj/structure/girder/broken(locate(x+1, y, z))
+	new /obj/structure/girder/broken(locate(x, y+1, z))
+	new /obj/item/stack/sheet/metal(locate(x+1, y+1, z), 10)
+	new /obj/item/stack/rods(locate(x+1, y+1, z), 20)
+
+	return ..()
+
+/obj/structure/watchtower/update_icon()
+	. = ..()
+	icon_state = "stage[stage]"
+
+	overlays.Cut()
+
+	if(stage >= WATCHTOWER_STAGE_BARRICADED)
+		overlays += image(icon=icon, icon_state="railings", layer=ABOVE_MOB_LAYER, pixel_y=25)
+
+	if (stage == WATCHTOWER_STAGE_COMPLETE)
+		roof_image = image(icon=icon, icon_state="roof", layer=ABOVE_MOB_LAYER, pixel_y=51)
+		roof_image.plane = ROOF_PLANE
+		roof_image.appearance_flags = KEEP_APART
+		overlays += roof_image
+
+/obj/structure/watchtower/attackby(obj/item/item, mob/user)
+	if(user.action_busy)
+		return
+	
+	if(istool(item) && !skillcheck(user, SKILL_CONSTRUCTION, SKILL_CONSTRUCTION_ENGI))
+		to_chat(user, SPAN_WARNING("You are not trained to configure [src]..."))
+		return TRUE
+
+	switch(stage)
+		if(WATCHTOWER_STAGE_WELDED)
+			if(!istype(item, /obj/item/stack/rods))
+				return
+
+			var/obj/item/stack/rods/rods = item
+
+			if(!do_after(user, 4 SECONDS * user.get_skill_duration_multiplier(SKILL_CONSTRUCTION), INTERRUPT_NO_NEEDHAND|BEHAVIOR_IMMOBILE, BUSY_ICON_FRIENDLY, src))
+				return
+
+			if(rods.use(60))
+				to_chat(user, SPAN_NOTICE("You add connection rods to the watchtower."))
+				stage = WATCHTOWER_STAGE_COLUMNS
+				update_icon()
+			else
+				to_chat(user, SPAN_NOTICE("You failed to construct the connection rods. You need more rods."))
```

Comment:
You either need to accumulate rods, or just immediately tell them before a do_after that they need all 60.